libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -1401,15 +1412,11 @@ void RCOutput::dshot_send(pwm_group &group, uint32_t time_out_us)
             uint16_t pwm = period[chan];
 
             if (safety_on && !(safety_mask & (1U<<(chan+chan_offset)))) {
-                // safety is on, overwride pwm
-                pwm = 0;
+                // safety is on, override pwm
+                continue;
             }
 
             const uint16_t chan_mask = (1U<<chan);
-            if (pwm == 0) {
```

Comment:
pwm could be 0 for reasons other than the safety mask.

Reply:
Setting pwm to 0 is not the right way to disable dshot. I can probably put this check back in, but it's not right.

Reply:
OK I have put it back in